src/raster/rasterband.rs

```diff
@@ -863,6 +954,50 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: i32,
+    histogram: *mut i32,
+    histogram_vec: Option<Vec<i32>>,
+}
+
+impl Histogram {
+    /// Histogram lower bound
+    pub fn min(&self) -> f64 {
+        self.min
+    }
+
+    /// Histogram upper bound
+    pub fn max(&self) -> f64 {
+        self.max
+    }
+    /// Histogram values for each bucket
+    pub fn histogram(&self) -> &[i32] {
```

Comment:
Consider naming `buckets` or `counts`? With this name, you'll end up with things like this:

```rust
let rb: Rasterband<'_> = ...;
let histogram = rb.get_histogram(...);

for count in histogram.histogram() {
...
}
```

Reply:
I also found it weird but I was just being consistent with GDAL naming. Happy to change it if we are not aiming for that.